src/lib/common/include/sol-platform.h

```diff
@@ -156,6 +160,75 @@ int sol_platform_restart_service(const char *service);
 
 int sol_platform_set_target(const char *target);
 
+struct sol_mount_point {
+    char *dev;
+    char *mpoint;
+    char *fstype;
+    unsigned long flags;
+    char *opts;
+    bool noauto;
+};
+
+static inline void
+sol_mount_point_free(struct sol_mount_point *mp)
+{
+    if (!mp)
+        return;
+
+    free(mp->dev);
+    free(mp->mpoint);
+    free(mp->fstype);
+    free(mp->opts);
+    free(mp);
+}
+
+/**
+ * List mounted mount points, if @c hplug_only is provided this function will list
+ * only mount points mounted by us, all mounted mount points are listed otherwise.
+ *
+ * @param hplug_only List only mount points only mounted by us?
+ * @param vector Initialized sol_vector where we store the resulting list
+ *
+ * @return 0 on success, negative errno otherwise
+ */
+int sol_platform_get_mount_points(bool hplug_only, struct sol_ptr_vector *vector);
+
+/**
+ * Umount a @c mpoint. If @c async_cb is provided the operation is run asynchronously
+ * in a spawned process and the @c async_cb function is called when @c mpoint
+ * is actually umounted.
+ *
+ * If the umount operation is desired to be run synchronously just provide @c async_cb
+ * as NULL.
+ *
+ * @param mpoint The mount point to be unmounted.
+ * @param async_cb The async callback to be called when async operation is required.
+ * @param data Context data to be provided to @c async_cb function.
+ *
+ * @return 0 on success, negative errno otherwise
+ */
+int sol_platform_umount(const char *mpoint, void (*async_cb)(const char *mpoint, void *data, uint64_t pid, int status), void *data);
```

Comment:
always take `const void *data`, to allow user given const pointers without cast.

and our convention is to always pass `void *data` (casted at callback time) as the first parameter to the callback function.

`pid` and `status` are not something that should be exposed in the documentation, they are implementation-dependent and generally as no interest to the user.

the documentation of this function should not mention anything about processes, it's an implementation detail. `async_cb` should always be given and it should receive the errno.

Proposed signature:

```c
int sol_platform_unmount(const char *mpoint, void (*cb)(void *data, const char *mpoint, int error), const void *data);
```

In linux-micro this will use fork-run and call `umount(2)` from the process, convert the result into exit code that is then returned by fork-run's `on_dead`, that is given to the user.

Not sure about systemd, but likely there is a way to request a mount operation via dbus, then you proceed like that and convert back the resulting error code, calling the user callback.

The `sol_platform_unmount()` error return is only if the implementation fail to start the async operation, like fork-run failing. The actual unmount operation error is given as callback's `error` parameter.